test/c_blog_server_test.js

```diff
@@ -0,0 +1,463 @@
+"use strict";
+
+const _       = require('lodash');
+const expect  = require('unexpected');
+const request = require('supertest');
+const baseUrl = 'http://localhost:3000';
+
+const User = require('../app/models/user');
+const Post = require('../app/models/post');
+const Comment = require('../app/models/comment');
+
+const bookshelf = require('../app/db/bookshelf');
+const app = require('../app/');
+
+let mockUser = {
+  name: 'Sally Low',
+  username: 'sally',
+  email: 'sally@example.org',
+  password: 'password',
+};
+
+let anotherMockUser = {
+  name: 'Bob Saltwater',
+  username: 'bob',
+  email: 'bob@example.org',
+  password: 'password',
+};
+
+let aThirdMockUser = {
+  name: 'Johnny Auslander',
+  username: 'johnny',
+  email: 'johnny@example.org',
+  password: 'password'
+};
+
+let mockPost = {
+  title: 'Test Post',
+  body: 'This is just a test post with no content.'
+};
+
+let anotherMockPost = {
+  title: 'Test Post 2',
+  body: 'This is a second post.'
+};
+
+let mockComment = {
+  body: 'This is just a test comment.'
+};
+
+const login = (server, opts) => {
```

Comment:
This is potentially a little scary, but I like that it's explicitly included in the file where it's used and not magically shimmed in from some hard-to-find `test/utils.js` or something :)